awsutil/awsutil_test.go

```diff
@@ -0,0 +1,76 @@
+package awsutil
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/stretchr/testify/require"
+)
+
+func TestECSTaskMeta(t *testing.T) {
+	taskRegion := "bogus-east-1"
+	nonTaskRegion := "some-other-region"
+	ecsMeta := ECSTaskMeta{
+		Cluster: "test",
+		TaskARN: fmt.Sprintf("arn:aws:ecs:%s:123456789:task/test/abcdef", taskRegion),
+		Family:  "task",
+	}
+
+	cases := []struct {
+		name         string
+		env          map[string]string
+		expectRegion string
+	}{
+		{
+			name:         "no-env",
+			env:          nil,
+			expectRegion: taskRegion,
+		},
+		{
+			name:         "aws-region",
+			env:          map[string]string{"AWS_REGION": nonTaskRegion},
+			expectRegion: nonTaskRegion,
+		},
+		{
+			name: "aws-default-region",
+			env: map[string]string{
+				// "AWS_DEFAULT_REGION is only read if AWS_SDK_LOAD_CONFIG is also set,
+				// and AWS_REGION is not also set."
+				"AWS_DEFAULT_REGION":  nonTaskRegion,
+				"AWS_SDK_LOAD_CONFIG": "1",
+			},
+			expectRegion: nonTaskRegion,
+		},
+		{
+			name: "aws-region-and-default-region",
+			env: map[string]string{
+				"AWS_REGION":          nonTaskRegion,
+				"AWS_DEFAULT_REGION":  "should-not-use-this-one",
+				"AWS_SDK_LOAD_CONFIG": "1",
+			},
+			expectRegion: nonTaskRegion,
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			os.Clearenv()
```

Comment:
would that clear all env variables even the ones we don't set? should we instead call `defer` to the env vars we're setting? If we would want to set env vars for tests at some point in the future, that could cause weird bugs that will take a while to debug.

Reply:
Yeah, this is a test isolation concern. Let me fix this.

Reply:
Okay, I addressed this in the newer with a `restoreEnv` function and defers (`t.Cleanup`, actually). I'm not sure I like it.

If we did drop all the tests of environment variables, we could remove this env cleanup stuff as well, and I'd be fine with that too.

Reply:
Oh I see. It's because somebody might have it in their env before running the test and we want to make sure it doesn't affect us and doesn't mutate someone's environment. 🤔 yeah, I'm not sure if there's a cleaner way than what you did. Generally, testing with env vars is kind of messy in my experience, so I'm OK with what you wrote.